app/code/community/SomethingDigital/EnterpriseIndexPerf/Model/Merchandiser.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+class SomethingDigital_EnterpriseIndexPerf_Model_Merchandiser
+{
+    /**
+     * newestFirst sort action.
+     *
+     * @param array $params Sorting parameters.
+     * @return void
+     */
+    public function newestFirst($params)
+    {
+        $catId = $params['catId'];
+        /** @var OnTap_Merchandiser_Model_Resource_Merchandiser $merchandiserResourceModel */
+        $merchandiserResourceModel = Mage::getResourceModel('merchandiser/merchandiser');
+
+        $categoryProducts = $merchandiserResourceModel->getCategoryProduct($catId, "product_id DESC");
+        $position = 1;
+        foreach ($categoryProducts as $product) {
```

Comment:
I can't wait until we can use collection pipelines for things like this - create a map of changed items and apply the update to that map instead of maintaining position counters, etc. :sigh: until that day comes...